test/kubernetes_unittest.cc

```diff
@@ -696,4 +699,137 @@ TEST_F(KubernetesTest, GetPodAndContainerMetadata) {
   EXPECT_EQ(pod_metadata->ToString(),
             m[2].metadata().metadata->ToString());
 }
+
+// Polls store until collected_at for resource is newer than
+// last_timestamp.  Returns false if newer timestamp not found after 3
+// seconds (polling every 100 millis).
+bool WaitForNewerCollectionTimestamp(const MetadataStore& store,
+                                     const MonitoredResource& resource,
+                                     Timestamp last_timestamp) {
+  for (int i = 0; i < 30; i++){
+    const auto metadata_map = store.GetMetadataMap();
+    const auto m = metadata_map.find(resource);
+    if (m != metadata_map.end() && m->second.collected_at > last_timestamp) {
+      return true;
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+  return false;
+}
+
+TEST_F(KubernetesTest, KubernetesUpdater) {
```

Comment:
Personal opinion here, it would be nice if this test were split into two tests, one for pods, and one for nodes.